src/librbd/journal/Types.h

```diff
@@ -121,12 +121,13 @@ struct OpFinishEvent : public OpEventBase {
 
 struct SnapEventBase : public OpEventBase {
   std::string snap_name;
+  cls::rbd::SnapshotNamespace snap_namespace;
```

Comment:
Only snap create and remove would require the namespace -- all other ops imply the user snapshot namespace. Either create a new SnapNamespaceBase for those two structs to inherit from or just add the snap namespace to those two ops. This comment applies to the notifications in WatchNotifyTypes as well.